docs/sources/setup/install/helm/deployment-guides/gcp.md

```diff
@@ -0,0 +1,625 @@
+{{< /admonition >}}
+
+## Create GCS buckets
+
+{{< admonition type="warning" >}}
+ **DO NOT** use the default bucket names;  `chunks`, `ruler` and `admin`. Choose a **unique** name for each bucket. For more information see the following [security update](https://grafana.com/blog/2024/06/27/grafana-security-update-grafana-loki-and-unintended-data-write-attempts-to-amazon-GCS-buckets/).
+{{< /admonition >}}
+
+Before deploying Loki, you need to create two GCS buckets: one to store logs (chunks) and another to store alert rules (ruler). You can create the bucket using the GCP Management Console or the GCP CLI. The bucket name must be globally unique.
+
+{{< admonition type="note" >}}
+GEL customers will require a third bucket to store the admin data. This bucket is not required for OSS users.
+{{< /admonition >}}
+
+```bash
+gcloud storage buckets create gs://<CHUNKS_BUCKET_NAME> gs://<RULER_BUCKET_NAME> \
+  --location=<REGION> \
+  --default-storage-class=STANDARD \
+  --public-access-prevention \
+  --uniform-bucket-level-access \
+  --soft-delete-duration=7d
+```
+
+Make sure to replace the `region` and `bucket` name with your desired values. We will revisit the bucket policy later in this guide.
+
+Here's an example with all the variables filled in:
+
+```bash
+gcloud storage buckets create gs://loki-gcp-chunks gs://loki-gcp-ruler \
+  --location=europe-west4 \
+  --default-storage-class=STANDARD \
+  --public-access-prevention \
+  --uniform-bucket-level-access \
+  --soft-delete-duration=7d
+```
+
+When you run this command, you should get something like this in response:
+
+```bash
+Creating gs://loki-gcp-chunks/...
+Creating gs://loki-gcp-ruler/...
+```
+
+
+## Defining IAM roles and policies
+
+IAM determines who can access which resources on GCP and can be configured in several ways. The recommended method for allowing Loki to access GCS is to use Workload Identity Federation. This method is more secure than using creating and distributing a service account key. The below steps show how to create the role and policy using the gcloud CLI.
+
+### Authenticating to the GKE cluster
+
+You need to be able to run `kubectl` commands on the cluster, so make sure you have it installed (run `gcloud components install kubectl` if not) and then run this command:
+
+```bash
+gcloud container clusters get-credentials <CLUSTER_NAME> \
+  --region=<REGION>
+```
+
+Here's an example of that command with the variables filled in:
+
+```bash
+gcloud container clusters get-credentials loki-gcp \
+  --region=europe-west4
+```
+
+This will authenticate you via your GCP IAM identity, write the cluster's access info to your local kubeconfig (usually `~/.kube/config`), and then allow `kubectl` commands talk to the right cluster from now on.
```

Comment:
```suggestion
This will authenticate you via your GCP IAM identity, write the cluster's access info to your local kubeconfig (usually `~/.kube/config`), and then allow `kubectl` commands to talk to the right cluster from now on.
```